lampod/src/ln/channel_manager.rs

```diff
@@ -448,10 +449,15 @@ impl ChannelEvents for LampoChannelManager {
         })
     }
 
-    fn close_channel(&self) -> error::Result<()> {
-        unimplemented!()
-    }
+    fn close_channel(&self, channel: request::CloseChannel) -> error::Result<()> {
```

Comment:
```suggestion
    fn close_channel(&self, node_id: &str, channel_id: &str) -> error::Result<()> {
```

Do not overcomplicate stuff, why we need a CloseChannel request here?

Reply:
We convert the given `node_id` and `channel_id` which is in string to the respective `PublicKey` and `ChannelId` which we need to pass to the `close_channel` function. These functions are implemented inside `model/close_channel.rs` which I think would be more suited there.
```rust
    fn close_channel(&self, channel: request::CloseChannel) -> error::Result<()> {
        let channel_id = channel.channel_id()?;
        let node_id = channel.counterpart_node_id()?;

        self.manager()
            .close_channel(&channel_id, &node_id)
            .map_err(|err| error::anyhow!("{:?}", err))?;
        Ok(())
    }
```

Also it follows the pattern that we have inside the `ChannelEvent` trait.
```rust
pub trait ChannelEvents {
    /// Open a Channel
    fn open_channel(
        &self,
        open_channel: request::OpenChannel,
    ) -> error::Result<response::OpenChannel>;

    /// Close a channel
    fn close_channel(&self, channel: request::CloseChannel) -> error::Result<()>;

    fn change_state_channel(&self, event: ChangeStateChannelEvent) -> error::Result<()>;
}
```

Reply:
Make sense, thanks for the explaination!